src/opera/pge/dswx_pge.py

```diff
@@ -180,20 +180,21 @@ def _core_filename(self, inter_filename=None):
         dataset_fields = get_hls_filename_fields(f"{dataset}.BAND.tif")
 
         source = dataset_fields['product']
-        sensor = get_geotiff_spacecraft_name(inter_filename)
+        spacecraft_name = get_geotiff_spacecraft_name(inter_filename)
         tile_id = dataset_fields['tile_id']
         timetag = dataset_fields['acquisition_time']
         version = dataset_fields['collection_version']
         subversion = dataset_fields['sub_version']
 
         # Assign the core file to the cached class attribute
         self._cached_core_filename = (
-            f"{self.PROJECT}_{self.LEVEL}_{self.NAME}_{source}_{sensor}_{tile_id}_{timetag}_{version}.{subversion}"
+            f"{self.PROJECT}_{self.LEVEL}_{self.NAME}_{source}_{spacecraft_name}_"
+            f"{tile_id}_{timetag}_{str(self.runconfig.product_counter).zfill(3)}_{version}.{subversion}"
```

Comment:
For consistency with NISAR, lets put the product counter at the very end, just prior to the file extension.

Reply:
completed